tella-locking-ui/src/main/java/com/hzontal/tella_locking_ui/ui/pin/pinview/CalculatorKeyView.kt

```diff
@@ -6,49 +6,150 @@ package com.hzontal.tella_locking_ui.ui.pin.pinview
+                    binding.plusMinusBtn.text = "()"
+                    binding.plusMinusBottomBtn.visibility = VISIBLE
+                    binding.zeroBtnGuideline.setGuidelinePercent(0.25F)
+                }
+
+                CalculatorTheme.ORANGE_SKIN -> {
+                    setButtonsGroupStyleWithBackgroundResource(binding.btnNumbersGroup, style.calculatorBackgroundDrawable, style.calculatorNumbersTextColor, 0)
+                    setButtonsGroupStyleWithBackgroundResource(binding.btnOperatorsGroup, style.calculatorBackgroundDrawable, style.calculatorOperatorsTextColor, 0)
+                    binding.okBtn.apply {
+                        setBackgroundResource(style.calculatorOkBtnTextBackgroundDrawable)
+                        setTextColor(ContextCompat.getColor(context, style.calculatorOkBtnTextColor))
+                    }
+                    binding.plusMinusBottomBtn.apply {
+                        setBackgroundResource(style.calculatorBackgroundDrawable)
+                        text = "0"
+                        visibility = VISIBLE
+                    }
+                    binding.plusMinusBtn.text = "()"
+                    binding.zeroBtn.text = "."
+                    binding.deleteBtn.text = "C"
+                    binding.commaBtn.setBackgroundResource(style.calculatorBackgroundDrawable)
+                    binding.zeroBtnGuideline.setGuidelinePercent(0.25F)
+
+                }
+
+                CalculatorTheme.YELLOW_SKIN -> {
+                    setButtonsGroupStyleWithBackgroundResource(binding.btnOperatorsGroup, style.calculatorBackgroundDrawable, style.calculatorOperatorsTextColor, 0)
+                    setButtonsGroupStyleWithBackgroundResource(binding.btnNumbersGroup, style.calculatorBackgroundDrawable, style.calculatorNumbersTextColor, 0)
+                    binding.plusMinusBottomBtn.visibility = GONE
+                    binding.commaBtn.apply {
+                        text = "."
+                        setBackgroundResource(style.calculatorBackgroundDrawable)
+                    }
+                    binding.okBtn.apply {
+                        setBackgroundResource(style.calculatorOkBtnTextBackgroundDrawable)
+                        setTextColor(ContextCompat.getColor(context, style.calculatorOkBtnTextColor))
+                    }
+                    binding.percentBtn.visibility = GONE
+                }
+                CalculatorTheme.GREEN_SKIN -> {
+                    setButtonsGroupMargin(binding.btnOperatorsGroup, 16)
+                    setButtonsGroupMargin(binding.btnNumbersGroup, 16)
+                    setTextViewMargin(binding.commaBtn, 16)
+                    setTextViewMargin(binding.okBtn, 16)
+                }
+            }
+
+    private fun setTextViewMargin(button: TextView, margin: Int) {
+        button.apply {
+            val params = layoutParams as ViewGroup.MarginLayoutParams
+            params.setMargins(margin)
```

Comment:
We already have `setMargins` extension  in the project, maybe after moving this class to the shared ui consider also moving the Extension class to the shared ui, and you can apply it directly.